python/src/tests/tasks/test_differential_expression.py

```diff
@@ -30,123 +39,154 @@ def get_request(
 
         return request
 
-    @pytest.fixture(autouse=True)
-    def load_correct_definition(self):
-        responses.add(
-            responses.POST,
-            f"{config.R_WORKER_URL}/v0/DifferentialExpression",
-            json={},
-            status=200,
-        )
-
     """
-    Mocks the S3 query for fetching cell sets. Returns an
-    empty cell set and yields the patched up object.
+    Returns a stubber and a stubbed s3 client that will get executed
+    in the code instead of the real s3 clients
```

Comment:
This is a better way to mock s3 calls -- rather than mocking the entire s3 helper, we only mock the actual calls that we make to s3 and use the real s3 helper. In this way, through these tests we also test the s3 helper.
Stubbers are a way to mock aws resources in Python. If you are interested, check this out: https://botocore.amazonaws.com/v1/documentation/api/latest/reference/stubber.html